tiledb/sm/buffer/buffer_list.cc

```diff
@@ -37,10 +37,11 @@
 namespace tiledb {
 namespace sm {
 
-BufferList::BufferList() {
-  current_buffer_index_ = 0;
-  current_relative_offset_ = 0;
-  offset_ = 0;
+BufferList::BufferList()
+    : buffers_{}
```

Comment:
One more comment, sorry: We only need to list-initialize 1) primitives (because they will otherwise be garbage) and 2) objects we wish to construct with something other than the default constructor.

The `buffers_` object falls into #2. In this scenario, we want the default constructor. I apologize for being pedantic, but we would prefer to remain consistent with how we list-init variables in the rest of the code base.

Reply:
No problem, already has been reverted back :)

Reply:
LGTM, thank you! I'll merge it once the CI passes.